### DISCUSSION TITLE
Static files not accessible - NextJS Plesk .js/.css extension changed to .jsundefined/.cssundefined

### DISCUSSION THREAD

**Discussion by Fauzdar1** [vercel/next.js#51170] [Help] [answered]
### Summary

I've a NextJS app deployed on Plesk server. It was working properly, I changed something in layout and pushed an update, since then, neither the new nor the backup is working.

The error states that the static files (JS and CSS) files are not available and gives a 404 error. The reason for this error is it's adding undefined at the end of all the URLs and I've no clue where is it coming from, I've found nothing similar to this one. Because of this, all CSS and JS files' extensions have changed to .cssundefined and .jsundefined, removing the undefined from the url works and the file's content opens in the browser.

I've been using the command next build and it was working with the same config properly. Also, the changes were only in a layout file, related to only few layout elements.

The same code is working properly at localhost, in both development and production mode. You can find the question I asked on StackOverflow [here](https://stackoverflow.com/q/76456405/8244632).

![Screenshot 2023-06-12 at 5 15 56 PM](https://github.com/vercel/next.js/assets/62703211/c5264f52-1dae-4d66-955b-43d8225f0d10)

![Screenshot 2023-06-12 at 5 25 47 PM](https://github.com/vercel/next.js/assets/62703211/7b37a872-ca1c-4943-9eb3-743f39c96297)

**Comment by icyJoseph** [comment]
What Node version is running on the deploy environment? I couldn't find anything in the webpack config that places content after the `ext` part. So, it is either calculating `ext` as `jsundefined` or `cssundefined` already, or it is done after the webpack step... but how... 

Also kind of weird that the fonts are not affected...

**Reply by Fauzdar1** [reply]
The NodeJS installed in the server is version 16.20.0, same version was used before as well. Version in the development machine is 16.18.1.

These are the dependancies used:

```
"dependencies": {
    "@aws-sdk/client-s3": "^3.303.0",
    "@aws-sdk/lib-storage": "^3.350.0",
    "@aws-sdk/s3-request-presigner": "^3.303.0",
    "@next-auth/mongodb-adapter": "^1.1.1",
    "@next/font": "^13.1.6",
    "@reduxjs/toolkit": "^1.6.2",
    "@stripe/react-stripe-js": "^1.16.1",
    "@stripe/stripe-js": "^1.46.0",
    "@styled-icons/bootstrap": "^10.47.0",
    "@styled-icons/boxicons-logos": "^10.47.0",
    "@styled-icons/material": "^10.47.0",
    "async": "^3.2.2",
    "axios": "^1.2.1",
    "bcrypt": "^5.0.1",
    "cross-env": "^7.0.3",
    "crypto-js": "^4.1.1",
    "custom-id-new": "^1.2.1",
    "dotenv": "^16.0.0",
    "express": "^4.18.2",
    "form-data": "^4.0.0",
    "formidable": "^2.1.1",
    "fuse.js": "^6.4.6",
    "html-to-image": "^1.11.3",
    "i18next": "^22.4.15",
    "js-cookie": "^3.0.1",
    "jspdf": "^2.5.1",
    "mongoose": "^6.8.1",
    "next": "^13.2.4",
    "next-auth": "^4.20.1",
    "next-redux-wrapper": "^7.0.5",
    "nextjs-progressbar": "^0.0.16",
    "ni18n": "^1.1.0",
    "node-fetch": "^3.3.0",
    "nodemailer": "^6.7.3",
    "react": "^18.2.0",
    "react-apexcharts": "^1.4.0",
    "react-colorful": "^5.5.1",
    "react-data-table-component": "^7.5.2",
    "react-dom": "^18.2.0",
    "react-i18next": "^12.2.2",
    "react-infinite-scroll-component": "^6.1.0",
    "react-inner-image-zoom": "^3.0.0",
    "react-modal": "^3.16.1",
    "react-multi-select-component": "^4.1.13",
    "react-redux": "^7.2.6",
    "react-responsive-carousel": "^3.2.22",
    "react-toastify": "^9.1.1",
    "sharp": "^0.31.3",
    "stripe": "^11.5.0",
    "styled-components": "^5.3.5",
    "suneditor": "^2.42.0",
    "suneditor-react": "^3.4.0",
    "swiper": "^8.4.5",
    "swr": "^2.0.0",
    "xss": "^1.0.14"
  },
  "devDependencies": {
    "eslint": "^8.36.0",
    "eslint-config-next": "^13.2.4"
  }
```

**Comment by alexilobyak** [comment]
Faced exact same issue today, but on docker env (node 18.14.2). Same version of next.

No critical code changes have been made today, just minor ui changes, and for last deploys I've got same errors..

**Reply by icyJoseph** [reply]
Could you try pinning the next.js version to 13.2.4? Assuming you have `^13.2.4` change it to just `13.2.4`

**Reply by icyJoseph** [reply]
Also, could you try to set up a reproduction repository? Since it's docker we could all quickly jump on it

**Comment by florinrelea** [comment]
Same problem here

**Reply by icyJoseph** [reply]
Can you please try to share or set up a repository? System specs, next version? Where is it and how is it getting deployed?

**Reply by florinrelea** [reply]
My next version is `13.4.1`. The app is deployed on AWS Elastic Beanstalk `Node.js 16 running on 64bit Amazon Linux 2/5.5.5`. 
I am trying to debug it now, will let you know if I can find the problem.

**Reply by florinrelea** [reply]
Fixed. This was a really strange error.
I changed `"next": "^13.4.1"` to `"next": "13.4.1"`.
I believe this error appeared after I installed `"sortablejs": "^1.15.0"` and `"react-color": "^2.19.3"`

**Reply by icyJoseph** [reply]
Aha so the floating dependency is the manifestation. 

So when you installed those packages, the floating dependency installed a later version of Next.js - 

Now the problem is... what has changed in that later version...

**Reply by icyJoseph** [reply, marked as answer]
I guess 13.4.5 has something broken for you guys. @Fauzdar1 @alexilobyak you also have to, for now, pin your dependency to 13.4.1 and regenerate your lock files.

Most likely going from 13.4.4 to 13.4.5 something has broken...

**Reply by icyJoseph** [reply]
@florinrelea do you also use styled components?

**Reply by florinrelea** [reply]
> @florinrelea do you also use styled components?

Yes

**Reply by icyJoseph** [reply]
Compared to https://github.com/vercel/next.js/tree/canary/examples/with-styled-components, how much difference do you have in the _document file?

**Reply by florinrelea** [reply]
This is my _document file. I am using `@mui/material` with styled components.
```
import { Html, Head, Main, NextScript } from 'next/document'
import Script from 'next/script'

export default function Document() {
  return (
    <Html lang="en">
      {
        process.env.NODE_ENV === 'production' ? <Head>
          {/*Global site tag (gtag.js) - Google Analytics */}
          <Script
            src="https://www.googletagmanager.com/gtag/js?id=foo"
            async
            strategy="afterInteractive"
          />

          <Script id="google-analytics" strategy="afterInteractive">
            {`
              window.dataLayer = window.dataLayer || [];
              function gtag(){window.dataLayer.push(arguments);}
              gtag('js', new Date());

              gtag('config', 'foo');
            `}
          </Script>
        </Head> : <Head />
      }

      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}

```

**Reply by icyJoseph** [reply]
MM false alarm I guess, but if you use styled components, you need to parse the styles as part of the SSR process. You can use the repo I shared above as reference.

Well no idea what else could it be... but I guess as people upgrade to 13.4.5, more and more instances of this bug should start to show up.

**Reply by Fauzdar1** [reply]
> I guess 13.4.5 has something broken for you guys. @Fauzdar1 @alexilobyak you also have to, for now, pin your dependency to 13.4.1 and regenerate your lock files.
> 
> Most likely going from 13.4.4 to 13.4.5 something has broken...

Thanks for this, working now. I guess reinstalling the dependancies had upgraded the version and it stopped working.

**Reply by icyJoseph** [reply]
I've tried to reproduce this in a linux machine and nothing. I guess we should brace ourselves for a bug wave, as people upgrade to 13.4.5 - there's a whole bunch of folks that are still in next@12

Do you guys have any special next.config.js setup? Or could you ever reproduce this locally by doing a `next build`, `next start` pass?

**Reply by Fauzdar1** [reply]
Couldn't able to reproduce the issue at local end, possibly because never installed the dependancies again in local machine and it was still on `13.2.4`.

But, here's my `next.config.js` if this helps:

```
const headers = require("./headers");
const next_config = {
  reactStrictMode: false,
  swcMinify: true,
  images: {
    remotePatterns: [
      {
        hostname: "**",
      },
    ],
  },
  i18n: {
    defaultLocale: "en",
    locales: ["en", "bn", "ar", "fr"],
  },
  async headers() {
    return [
      {
        source: "/(.*)",
        headers,
      },
    ];
  },
  env: {
    APP_ENV: process.env.NODE_ENV,
    APP_SSL: process.env.NEXT_PUBLIC_APP_SSL,
  },
};

module.exports = next_config;
```

Also, tried that trailingSlash and replace-in-file solution, for just in case.

**Reply by alexilobyak** [reply]
For me, migration from '13.2.4' to '13.2.5' fixed issue.

**Reply by florinrelea** [reply]
I really can't reproduce it but this is something close to what I have: https://github.com/florinrelea/nextjs-bug-repo
